src/v/redpanda/admin_server.cc

```diff
@@ -4521,11 +4521,14 @@ void admin_server::register_cluster_routes() {
             .then([](auto health_overview) {
                 ss::httpd::cluster_json::cluster_health_overview ret;
                 ret.is_healthy = health_overview.is_healthy;
+
+                ret.unhealthy_reasons._set = true;
```

Comment:
This pattern is so that we also encode it into the JSON if the list itself is empty right?

Reply:
Exactly. I also had to double check this. Basically the assignment operator for a json list type doesn't set the property as "used" unless it adds at least 1 element, which I sort of feel is a bug (it just falls out of the way it calls push back), so you do this weird thing of setting `_set` manually.